docs/vec_of_option.md

```diff
@@ -0,0 +1,212 @@
+
+# 12. `Vec<Option<T>>` vs. `Vec<T>`
+
+> "I got, I got, I got, I got options" – _Pitbull_, before writing his polars plugins
+
+One situation you might encounter when developing plugins or working with polars in rust is to decide whether to return a `Vec<T>` or a `Vec<Option<T>>`.
+A `Vec<Option<T>>` often seems like a good idea, as it's able to represent __invalid__ values without you having to resort to hacky workarounds.
+However, it does take its toll. It's easy to see how with a small example, not even related to polars:
+
+```rust
+use std::mem::size_of_val;
+
+fn main() {
+    let vector: Vec<i32> = vec![1, 2, 3];
+    println!("{}", size_of_val(&*vector));
+    // Output: 12
+
+    let vector: Vec<Option<i32>> = vec![Some(1), Some(2), Some(3)];
+    println!("{}", size_of_val(&*vector));
+    // Output: 24
+}
+```
+
+For `i32` that's twice the memory usage, yikes!
+So you might be wondering: how could I avoid that overhead?
+
+
+## Validity mask
+
+If we look close at some polars functions, we can see they get around this in different ways.
+One such way can be seen in `interpolate_impl`:
```

Comment:
perhaps include a link to the Polars Series interpolate docs, saying that this is the Rust function which does the heavy lifting for it?

Reply:
Done!